### PR TITLE
doc: add doc for selective slashing

### PR DESCRIPTION
**Pull request by SebastianElvis**

### CODE REVIEW

File: x/btcstaking/README.md [hunk 1]
```diff
@@ -633,6 +634,52 @@ message MsgUpdateParams {
 }
 ```
 
+### MsgSelectiveSlashingEvidence
+
+The `MsgSelectiveSlashingEvidence` message is used for submitting evidences for
+selective slashing offences. In a selective slashing offence, the adversarial
+finality provider chooses a victim BTC delegation, signs its slashing
+transaction and decrypts covenant adaptor signatures to the Schnorr signatures
+using its secret key, then submits this slashing transaction to Bitcoin. By
```
Comment (vitsalis) [inline x/btcstaking/README.md:643]:
```suggestion
transaction, and decrypts covenant adaptor signatures to the Schnorr signatures
using its secret key before submitting this slashing transaction to Bitcoin. By
```

File: x/btcstaking/README.md [hunk 2]
```diff
@@ -633,6 +634,52 @@ message MsgUpdateParams {
...
+observing a pair of Schnorr signature and adaptor signature from covenant
```
Comment (vitsalis) [inline x/btcstaking/README.md:644]:
```suggestion
observing a pair of a Schnorr signature and an adaptor signature from the covenant
```

File: x/btcstaking/README.md [hunk 3]
```diff
@@ -633,6 +634,52 @@ message MsgUpdateParams {
...
+committee, anyone can extract the finality provider's secret key via the
+[adaptor signature](../../crypto/schnorr-adaptor-signature/README.md).
```
Comment (vitsalis) [inline x/btcstaking/README.md:646]:
```suggestion
committee, anyone can extract the finality provider's secret key due to the
[adaptor signature properties](../../crypto/schnorr-adaptor-signature/README.md).
```

File: x/btcstaking/README.md [hunk 4]
```diff
@@ -633,6 +634,52 @@ message MsgUpdateParams {
...
+2. Ensure the BTC delegation is active or unbonding.
+3. Ensure the given secret key corresponds to the finality provider's public
+   key.
+4. At this point, the finality provider must have done selective slashing. Thus,
+   slash the finality provider and emit an event `EventSelectiveSlashing` about
+   this.
+
+The `MsgSelectiveSlashingEvidence` is typically reported by the [BTC staking
+tracker](https://github.com/babylonchain/vigilante/tree/dev/btcstaking-tracker)
+program. It keeps monitoring slashing transactions on Bitcoin. Upon each
```
Comment (vitsalis) [inline x/btcstaking/README.md:678]:
```suggestion
program. It keeps monitoring for slashing transactions on Bitcoin. Upon each
```

**Review by vitsalis — APPROVED** [review]
Nice!

File: x/btcstaking/README.md [hunk 5]
```diff
@@ -633,6 +634,52 @@ message MsgUpdateParams {
 }
 ```
 
+### MsgSelectiveSlashingEvidence
+
+The `MsgSelectiveSlashingEvidence` message is used for submitting evidences for
+selective slashing offences. In a selective slashing offence, the adversarial
+finality provider chooses a victim BTC delegation, signs its slashing
+transaction and decrypts covenant adaptor signatures to the Schnorr signatures
```
Comment (gitferry) [inline x/btcstaking/README.md:642]: maybe add a reference to adaptor signatures and Schnorr signatures?

**Review by gitferry — APPROVED** [review]
Lgtm!

**Pull request merged**